.github/workflows/repo2docker.yml

```diff
@@ -15,7 +15,27 @@ jobs:
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-python@v2
-      - name: repo2docker install
-        run: pip install --upgrade jupyter-repo2docker
-      - name: Build container
-        run: repo2docker --no-run .
+      - name: Set no_push variable
```

Comment:
Wonder if just using
```
run: repo2docker --no-run .
if: no-tag
uses: jupyterhub/repo2docker-action@master
with:
if: tag
```
wouldn't be more maintainable.

Reply:
This will means installing`` jupyter-repo2docker`` manually (as we had before)
or using the docker container installed by ``jupyterhub/repo2docker-action``. I am not convinced it will simplify thins. In this case we only one approach allowing to ``build`` or ``build and push``

Reply:
I was thinking mainly in terms of total numbers of lines. If your new block is only run on tags, then it's just one extra line, right?

Reply:
what will need to be added will be
```
- name: Get the version
        id: get_version
        run: echo ::set-output name=VERSION::${GITHUB_REF#refs/tags/}
      - name: update jupyter dependencies with repo2docker
        uses: jupyterhub/repo2docker-action@master
        if: startsWith(github.ref, 'refs/tags')
        with:
          DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
          DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
          PUBLIC_REGISTRY_CHECK: true
          ADDITIONAL_TAG: ${{ steps.get_version.outputs.VERSION }}
          IMAGE_NAME: ${{ github.repository }}
```
or 
```
- name: update jupyter dependencies with repo2docker
        uses: jupyterhub/repo2docker-action@master
        if: startsWith(github.ref, 'refs/tags')
        with:
          DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
          DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
          PUBLIC_REGISTRY_CHECK: true
          IMAGE_NAME: ${{ github.repository }}
```
if we want to have the commit sha cf. ``https://hub.docker.com/layers/135963980/jburel/omero-guide-python/daa95a38d4bf/images/sha256-fe81ab02ded8300a117054668597deddde6f6cd181230f25d5ee4c0b5101a977?context=explore
(I don't think that is a good option if we push on tag)
```
that's why I am not sold on the fact that it will make things more maintable

Reply:
I could remove 
```
      - name: Check no no_push_to_cache
        run: echo ${{ steps.nopush.outputs.nopush }}
```
since it is only there to be sure it is set correctly

Reply:
removed